cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -443,8 +443,9 @@ struct DefaultExtensionIdRegistry : ExtensionIdRegistryImpl {
     // ARROW-15535.
     for (util::string_view name : {
              "add",
-             "equal",
-             "is_not_distinct_from",
+             "equal",                 // added to support join operator
```

Comment:
The comments are probably not needed as this will change significantly in the function mapping PR but they don't hurt.